crates/lib/Cargo.toml

```diff
@@ -59,6 +59,11 @@ zmq = { version = "0.10.0" }
 itertools = "0.10.5"
 rstest = "0.17.0"
 insta = "1.29.0"
+derive_builder = "0.12.0"
+
+# Not used directly by the crate, but required to avoid using yanked versions of crates that trigger
+# `cargo-deny` errors
+hermit-abi = "^0.3.2"
```

Comment:
0.3.1 was yanked and made `cargo-deny` throw an error. I assume we don't want to maintain `Cargo.lock` in this repo, since it is library (though it is tracked currently, I'll create a follow-up issue to remove it), so is this how we should handle it? There is no way to declare an exception for specific yanked crates.

I'm not sure if this makes sense, since it doesn't allow users to use crates below `0.3.2` that weren't necessarily yanked. Alternatively, we could reduce the severity of yanked crates to `warn`, which is [the default](https://embarkstudios.github.io/cargo-deny/checks/advisories/cfg.html#the-yanked-field-optional).

Reply:
Per a recommendation by Kalan, I tried to see if we could have a more flexible bounds that excludes `0.3.1` by specifying multiple ranges like you can in [npm](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#dependencies). Unfortunately, [the cargo book] suggests that it has no such equivalent.

Reply:
We talked about having a lock-file in libraries used by Python [here](https://rigetti.slack.com/archives/G011GB25W0K/p1687367116016159) a few weeks ago... looks like we chose to ignore the lockfile in `quil-rs`.

Won't this self-resolve any time a downstream Cargo build is done without `--locked`?

Reply:
Yeah, I tried to be clever and not rely on changing the lockfile since we were going to remove it, but it is the reason the yanked version was being chosen. I reverted this change and just updated the lockfile. I'll leave the removal to #319.